src/DIRAC/ConfigurationSystem/Client/test/Test_LocalConfiguration.py

```diff
@@ -0,0 +1,218 @@
+""" Unit tests for PathFinder only for functions that I added
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import mock
+import pytest
+import random
+
+from DIRAC import S_OK, S_ERROR
+import DIRAC.ConfigurationSystem.Client.LocalConfiguration
+
+
+doc_description = """
+This is an example of a typical team description.
+
+With indents and explanatory inserts::
+
+  * possible option
+  * possible option
+
+UnprocessedUnit:
+  Some interesting information
+
+"""
+doc_usage = """
+Usage:
+  my-script [options] ...
+"""
+doc_gen_opts = """
+General options:
+  -o  --option <value>         : Option=value to add
+  -s  --section <value>        : Set base section for relative parsed options
+  -c  --cert <value>           : Use server certificate to connect to Core Services
+  -d  --debug                  : Set debug mode (-ddd is extra debug)
+  -   --cfg=                   : Load additional config file
+  -   --autoreload             : Automatically restart if there's any change in the module
+  -   --license                : Show DIRAC's LICENSE
+  -h  --help                   : Shows this help
+"""
+doc_options = """
+Options:
+  -H  --host <value>  : host
+  -V  --vo <value>    : vo
+"""
+doc_arguments = """
+Arguments:
+  info:  some info
+"""
+doc_example = """
+Example:
+  $ my-script -H dirac
+  Hello dirac
+"""
+
+doc_blocks = [doc_usage, doc_options, doc_gen_opts, doc_arguments, doc_example]
+
+opts = ()
+args = []
+output = []
+result = None
+exitCode = None
+
+
+def mock_notice(*a):
+  global output
+  output.append(' '.join(a))
+
+
+def mock_exit(c=0):
+  global exitCode
+  exitCode = c
+
+
+def mock_gnu(*a):
+  global opts, args
+  return opts, args
+
+
+def action(a):
+  global result
+  result = a if a else 'no'
+  return S_OK()
+
+
+@pytest.fixture
+def localCFG():
+  with mock.patch('DIRAC.ConfigurationSystem.Client.LocalConfiguration.gLogger.notice', side_effect=mock_notice):
```

Comment:
I think you can use the [`monkeypatch` fixture](https://docs.pytest.org/en/stable/monkeypatch.html) to avoid needing lots of nested `with` statements.

Reply:
ok